subsys/mgmt/mcumgr/smp_shell.c

```diff
@@ -138,8 +142,15 @@ static uint16_t smp_shell_get_mtu(const struct net_buf *nb)
 
 static int smp_shell_tx_raw(const void *data, int len, void *arg)
 {
-	/* Cast away const. */
-	k_str_out((void *)data, len);
+	const struct shell_uart *const su = shell_backend_uart_get_ptr();
+	const struct shell_uart_ctrl_blk *const scb = su->ctrl_blk;
+	const uint8_t *out = data;
+
+	while (out && len) {
```

Comment:
Some day real soon this would have to become:

    while ((out != NULL) && (len > 0)) {

since implicit treatment of pointers and integers as booleans is a MISRA violation.

Reply:
Will fix.